tools/ci_connector_ops/ci_connector_ops/pipelines/commands/metadata_service.py

```diff
@@ -0,0 +1,66 @@
+import sys
+import click
+import anyio
+import dagger
+import logging
+
+from ci_connector_ops.pipelines.actions import environments
+from ci_connector_ops.pipelines.contexts import PipelineContext
+from ci_connector_ops.pipelines.utils import (
+    DAGGER_CONFIG,
+    with_exit_code,
+)
+from rich.logging import RichHandler
+
+
+logging.basicConfig(level=logging.INFO, format="%(name)s: %(message)s", datefmt="[%X]", handlers=[RichHandler(rich_tracebacks=True)])
+logger = logging.getLogger(__name__)
+
+METADATA_LIB_MODULE_PATH = "airbyte-ci/connectors/metadata_service/lib"
+
+
+async def run_metadata_lib_test_pipeline(metadata_pipeline_context):
+    async with dagger.Connection(DAGGER_CONFIG) as dagger_client:
+        metadata_pipeline_context.dagger_client = dagger_client.pipeline(metadata_pipeline_context.pipeline_name)
+        async with metadata_pipeline_context:
+            metadata_lib_module = environments.with_poetry_module(metadata_pipeline_context, METADATA_LIB_MODULE_PATH)
+            exit_code = await with_exit_code(metadata_lib_module.with_exec(["poetry", "run", "pytest"]))
+
+            # Raise an exception if the exit code is not 0
+            if exit_code != 0:
```

Comment:
@alafanechere I didnt like that this was where I ended up. Do you have any ideas as to how we can make the pipeline context own the handling of the status code?

For the record the reason I raise is to take advantage of the git error status reporting in `__aexit__` inside the pipeline class

Reply:
I would suggest you implement a `Step` class like `MetadataLibTest(Step)` and make its `_run` return  `self.get_step_result(metadata_lib_module)` .
Like what I've done for [QaChecks](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/checks.py#L12)

Reply:
I would also suggest you make use of [ConnectorTestReport](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/bases.py#L118) (or a Metadata version of it). This will help normalize the way we report success or failure to users.